server/app/com/xsn/explorer/services/CurrencyService.scala

```diff
@@ -60,45 +69,36 @@ class CurrencyServiceCoinMarketCapImpl @Inject()(
     }
   }
 
-  override def getUSDPrice: FutureApplicationResult[BigDecimal] = {
-    retrying {
-      requestFor(s"v1/tools/price-conversion?id=${coinMarketCapConfig.coinID.string}&amount=1&convert=USD").get().map {
-        response =>
-          (response.status, response) match {
-            case (200, r) =>
-              Try(r.json).toOption
-                .map { json =>
-                  (json \ "data" \ "quote" \ "USD" \ "price")
-                    .asOpt[BigDecimal]
-                    .map(Good(_))
-                    .getOrElse(Bad(One(CoinMarketCapUnexpectedResponseError)))
-                }
-                .getOrElse(Bad(One(CoinMarketCapUnexpectedResponseError)))
-            case (code, _) =>
-              Bad(One(CoinMarketCapRequestFailedError(code)))
-          }
-      }
-    }
-  }
+  override def getPrices: FutureApplicationResult[Map[Currency, BigDecimal]] = {
```

Comment:
Just add a single method like `getPrice(currency)` that knows how to retrieve a single price, so that other callers can get only what they need, a problem with the current approach is that it retrieves everything or nothing.

Reply:
with this you mean, keep both `getPrices` and `getPrice(currency)` or just have `getPrice(currency)` and get rid of `getPrices`?